fix: ignore undefined values from  array

sushantdhiman opened the pull request:
Closes https://github.com/goodybag/mongo-sql/issues/163

`$in` will ignore `undefined` values now. As for `null` values there is no point in crossing them to single `null` until you got insane number of `null` in that array :D

jrf0110 commented:
This actually wouldn't completely address #163 - 

> 2. null values should be joined with `field1 in (value1, value2, ...) or field1 is null`

Might be good to just merge this and we can do another PR for the second item

sushantdhiman commented:
I thought `$in [null]` would cover `column is null` case, but I have made changes to append `is` when `null` is detected in set.

Also refactored `$nin` so we dont duplicate same logic in `$in` and `$nin`, added some tests to prevent regressions

jrf0110 commented:
@sushantdhiman awwwwesome. Thanks a lot!

jrf0110 commented:
Since this changed the output of an existing helper, I've published as a major version bump `mongo-sql@5.0.0`
